state/util.go

```diff
@@ -0,0 +1,21 @@
+package state
```

Comment:
I wonder if we shouldn't put this into something like`util/oid_map.go` instead, since there might be use cases for this outside of the direct state logic (though `Oid` is defined in the `state` package right now)

Reply:
That makes sense but I feel like in a smaller codebase with a limited number of active contributors, it's fine to leave refactoring like that until you need it: I don't think either of us are likely to forget that this helper exists and accidentally reimplement it somewhere else. If we do need it elsewhere, we can move it (and Oid) before tackling that without any more trouble.